File: CondominiumAlerts.Backend/CondominiumAlerts.Features/Features/Messages/ChatMessageDto.cs
using CondominiumAlerts.Domain.Aggregates.Entities;

namespace CondominiumAlerts.Features.Features.Messages;

public record ChatMessageDto(
    Guid Id,
    string Text,
    Guid? CondominiumId,
    string? ReceiverUserId,
    Guid? MessageBeingRepliedToId,
    DateTime CreatedAt,
    UserDto CreatorUser,
    string? MediaUrl);
    

File: CondominiumAlerts.Frontend/src/app/core/models/chatMessage.dto.ts
import { ChatCreatorUserDto } from "./chatCreatorUser.dto";

export type ChatMessageDto = {
    id: string;
    text: string;
    mediaUrl?: string;
    creatorUser: ChatCreatorUserDto;
    creatorUserId?: string;
    receiverUserId?: string;
    condominiumId?: string;
    messageBeingRepliedToId?: string;
    createdAt: Date;
}